db/migrate/20200520124359_add_api_key_scopes.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class AddApiKeyScopes < ActiveRecord::Migration[6.0]
+  def change
+    create_table :api_key_scopes do |t|
+      t.integer :api_key_id
+      t.string :resource
```

Comment:
What values can `resource` be? If it is from a fixed set, it may be more efficient to store an integer representation rather than a string. I think it'll be the same for `action` as well.

Reply:
I suspect this gets tricky because plugins are allowed to contribute new resources/scopes, so we can't pre-allocate consistent integer ids. I think a string is more flexible in this case. This table is never going to a be a performance bottleneck, so I think we can take the hit.